engine/src/main/java/org/camunda/bpm/engine/impl/cmd/GetTelemetryDataCmd.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl.cmd;
+
+import org.camunda.bpm.engine.impl.interceptor.Command;
+import org.camunda.bpm.engine.impl.interceptor.CommandContext;
+import org.camunda.bpm.engine.impl.telemetry.dto.TelemetryData;
+
+public class GetTelemetryDataCmd implements Command<TelemetryData> {
+
+  @Override
+  public TelemetryData execute(CommandContext commandContext) {
+    return commandContext.getProcessEngineConfiguration().getTelemetryData();
+  }
```

Comment:
This implementation won't return any metrics or command counts. Please review `TelemetrySendingTask#run` to see how the sending tasks collects and sends the data.

On a related note: The test case doesn't find this because it does not test the feature end-to-end but only your implementation assumption (i.e. the assumption that the engine configuration is the central collection for all metrics data). I suggest to write a test that actually creates metric data etc. and not work with mocked data.